Update contributing docs (#2387)
knative/docs: CONTRIBUTING.md

**First off, thanks for taking the time to contribute!**

The following are guidelines for contributing to the Knative documentation.
These are just guidelines, not rules. Use your best judgment, and feel free to
propose changes to this document in a pull request.

## Before you begin

### Code of conduct

Knative follows the [Knative Code of Conduct](../CODE-OF-CONDUCT.md). By
participating, you are expected to uphold this code. Please report unacceptable
behavior to knative-code-of-conduct@googlegroups.com.

### Contributor license agreements

We'd love to accept your contributions! But before we can take them, you need to
fill out the [Google CLA](https://cla.developers.google.com). Important: You
must fill out the CLA with the same email address that you used to create your
GitHub account.

Once you are CLA'ed, we'll be able to accept your pull requests. This is
necessary because you own the copyright to your changes, even after your
contribution becomes part of this project. So this agreement simply gives us
permission to use and redistribute your contributions as part of the project.

Note: Your contributions are verified using the email address for which you

> use to sign the CLA. Therefore, 

Reviewer: <!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
use to sign the CLA. Therefore,
```

[setting your GitHub account to private](https://help.github.com/en/articles/setting-your-commit-email-address)

> is unsupported because all commits from private accounts are sent from the `noreply` 

Reviewer: <!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
is unsupported because all commits from private accounts are sent from the `noreply`
```

meets weekly on Tuesdays and alternates between a 9am PT and a 4:30pm PT time to
accommodate contributors in both the EMEA and APAC timezones.
[Click here](https://calendar.google.com/calendar/embed?src=google.com_18un4fuh6rokqf8hmfftm5oqq4%40group.calendar.google.com)
to see the exact dates on the Knative working group calendar. If you're
interested in becoming more involved in Knative's documentation, start attending
the working group meetings. You'll meet the writers currently steering our
documentation efforts, who are happy to help you get started.

### Getting started

There are a couple different ways to jump in to the Knative doc set:

- Look for a
  [Good First Issue](https://github.com/knative/docs/labels/kind%2Fgood-first-issue)
  in the backlog.

- Try out Knative and send us feedback. For example, run through one of the
  [install guides](docs/install/README.md) and then try
  [Getting Started with Knative Serving](docs/serving/getting-started-knative-app.md).

  You should keep a
  [friction log](https://devrel.net/developer-experience/an-introduction-to-friction-logging)
  of your experience and then use that to open your first set of PRs. Examples:

>   - What was hard for you?

Reviewer 2: maybe 'difficult' instead of 'hard', but that's me nitpicking :slightly_smiling_face:

Author: Good point, thanks!

>   - Did you stumble on something because it wasn't clear?

Reviewer 2: rather than 'it', maybe something like 'because the steps weren't clear'?

  - Was a dependency not mentioned?

**A few pointers and other considerations**

> - Think about how you can improve the documentation as a whole. <br> For

Reviewer 2: Is the `<br>` required here?

Author: Removed.

  example, how can you fix the issue you found so that others don't run into the
  same challenges?

> - Are there multiple places that could be fixed?

Reviewer 2: `that could be fixed` -> `where the issue could be fixed`?

Author: Yes. (Feel free to use "suggest" in GitHub if you want; I think the CLA-bot has finally sorted that out.)

>   - Are there other pages that you should apply your update?

Reviewer 2: that -> where?

Author: Rewrote this to suggest making sure that all pages related to the topic are updated.

>   - Would it help if there was a link to more or related information on a the

Reviewer 2: typo, "on a the page" -> "on a page"

    page? On a related page?

- Was the title or description misleading? Did you expect to find something

>   else?

Reviewer 2: maybe "find something else in a particular docs section"?

Author: Rewrote -- I think the idea was "did the title/navigation cause you to be surprised about the content of the page"

  possible for certain reasons, for example other commitments and time
  constraints.

While a sample might be out of date, it could still provide assistance and help
you get up-and-running with certain use-cases. If you find that something is not
right or contains outdated info, open an
[Issue](https://github.com/knative/docs/issues/new). The sample might be fixed
if bandwidth or available resource exists, or the sample might be taken down and
archived into the last release branch where it worked.

#### Choosing the correct branch

It is likely that your docs contribution is either for new or changed features
in the product, or for a fix or update existing content.

- **New or changed features**: If you are adding or updating documentation for a
  new or changed feature, you likely want to open your PR against the `master`
  branch. All pre-release content for active Knative development belongs in
  [`master`](https://github.com/knative/docs/tree/master/).

- **Fixes and updates**: If you find an issue in a past release, for example a
  typo or out-of-date content, you likely need to open multiple and subsequent

>   PRs. If not a followup PR, at least add the "`cherrypick` labels" to your

Reviewer 2: I don't think most users have permissions to add labels?

Author: I don't recall, but we should be able to use some prow `/label` commands or the like to manage this. (And it would be really nice to have Prow be able to auto-cherrypick changes to release branches.)

  original PR to indicate in which of the past release that your change affects.

  For example, if you find a typo in a page of the `v0.5` release, then that
  page in the `master` branch likely also has that typo.

  To fix the typo:

  1.  Open a PR against the
      [`master`](https://github.com/knative/docs/tree/master/) branch.
  1.  Add one or more `cherrypick-#.#` labels to that PR to indicate which of
      the past release branches should also be fixed. Generally, we only
      maintain the most recent numbered release.
  1.  If you want to complete the fix yourself (**best practice**), you then open a
      subsequent PR by running `git cherry-pick [COMMIT#]` against the
      `release-0.5`.
      Where  `[COMMIT#]` is the commit of the PR that you merged in `master`.

      Note: Depending on workload and available bandwidth, one of the Knative
      team members might be able to help handle the `git cherry-pick` in order
      to push the fix into the affected release branch(es).

> See a list of the available documentaion versions in the

Reviewer: <!--typo-bot-->
Found potential misspelling(s):
```suggestion
See a list of the available documentation versions in the
```

[branches page](https://github.com/knative/docs/branches)
of the `knative/docs` repo.

## Assigning owners and reviewers

> For both documentation and code samples, you should assign your PR to a single

Reviewer 2: likewise, IDK if most users have permissions to assign PRs, etc?

Author: This should be possible with `/assign`

[Knative Code of Conduct](https://github.com/knative/community/blob/master/CODE-OF-CONDUCT.md).
By participating, you are expected to uphold this code. Please report
unacceptable behavior to knative-code-of-conduct@googlegroups.com.

### Contributor license agreements

We'd love to accept your contributions! But before we can take them, you need to
fill out the [Google CLA](https://cla.developers.google.com). Important: You
must fill out the CLA with the same email address that you used to create your
GitHub account.

Once you are CLA'ed, we'll be able to accept your pull requests. This is
necessary because you own the copyright to your changes, even after your
contribution becomes part of this project. So this agreement simply gives us
permission to use and redistribute your contributions as part of the project.

Note: Your contributions are verified using the email address for which you use
to sign the CLA. Therefore,
[setting your GitHub account to private](https://help.github.com/en/articles/setting-your-commit-email-address)
is unsupported because all commits from private accounts are sent from the
`noreply` email address.

### Style guide

Knative documentation follows the

> [Google Developer Documentation Style Guide](https://developers.google.com/style/);

Reviewer 2: ```suggestion
[Google Developer Documentation Style Guide](https://developers.google.com/style/). Use this as a reference for writing style questions.
```

use it as a reference for writing style questions.

## Contributing to the documentation

### Reporting documentation issues

Knative uses Github issues to track documentation issues and requests. If you
see a problem with the documentation, submit an issue using the following steps:

1. Check the [Knative docs issues list](https://github.com/knative/docs/issues)
   before creating an issue to avoid making a duplicate.
2. Use the [correct template](https://github.com/knative/docs/issues/new) for
   your new issue. There are two templates available:
   - **Bug report**: If you're reporting an error in the existing documentation,

>      use this template. This could be anything from broken samples to typos.

Reviewer 2: What is a `broken sample`? Should this maybe be `broken link` instead?

Author: A broken sample would presumably be one of the code samples which no longer compiles or works properly due to ecosystem changes.

Author: A broken sample would presumably be one of the code samples which no longer compiles or works properly due to ecosystem changes.

     When you create a bug report, include as many details as possible and
     include suggested fixes to the issue. If you know which Knative component
     your bug is related to, you can assign the appropriate
     [Working Group Lead](https://github.com/knative/community/blob/master/working-groups/WORKING-GROUPS.md).
   - **Feature request**: For upcoming changes to the documentation or requests
     for more information on a particular subject.

Note that code issues should be filed against the
[individual Knative repositories](http://github.com/knative), while
documentation issues should go in the
[`knative/docs`repo](https://github.com/knative/docs/issues). If the issue is
specific to the [https://knative.dev](https://knative.dev) website, open the
issue in the [`knative/website`repo](https://github.com/knative/website/issues).

### Working group

The
[Knative Documentation Working Group](https://github.com/knative/community/blob/master/working-groups/WORKING-GROUPS.md#documentation)

> meets biweekly on Tuesdays at 9:30am PT. 

Reviewer: <!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
meets biweekly on Tuesdays at 9:30am PT.
```

> accommodate contributors in both the EMEA and APAC timezones.

Reviewer 2: The meeting is only at 9am now I think (11.30am CDT)?

Author: 9:30am, corrected. We seem to have lost the afternoon meeting that was the other weeks. Given the current level of WG activity, this seems reasonable. Updated.

[Click here](https://calendar.google.com/calendar/embed?src=google.com_18un4fuh6rokqf8hmfftm5oqq4%40group.calendar.google.com)

> to see the exact dates on the Knative working group calendar. If you're

Reviewer 2: Maybe split this into a new paragraph to make it more readable?

Author: Done.

interested in becoming more involved in Knative's documentation, start attending
the working group meetings. You'll meet the writers currently steering our
documentation efforts, who are happy to help you get started.

### Getting started

> There are a couple different ways to jump in to the Knative doc set:

Reviewer 2: ```suggestion
There are a couple of different ways to get started with contributing to the Knative documentation:
```

    samples that are contributor focused also belong in their corresponding
    component's repo. For example, Eventing specific test code is located in the
    [knative/eventing tests](https://github.com/knative/eventing/tree/master/test)
    folder.

- **User-focused content**

  - _Documentation_: Includes all content for Knative users. The external-facing
    user documentation belongs in the
    [`knative/docs` repo](https://github.com/knative/docs). All content in
    `knative/docs` is published to [https://knative.dev](https://knative.dev).

  - _Code samples_: Includes user-facing code samples and their accompanying
    step-by-step instructions. User code samples are currently separated into
    two different locations within the `knative/docs` repo. See the following
    section for details about determining where you can add your code sample.

##### Determining where to add user focused code samples

There are currently two categories of user-focused code samples, _Knative owned
and maintained_ and _Community owned and maintained_.

- **Knative owned and maintained**: Includes code samples that are actively

>   maintained and e2e tested. To ensure content currency and balance available

Reviewer 3: ```suggestion
  maintained and e2e tested. To ensure content is current and balance available
```

content currency may be confusing to non-native speakers

>   resource, only the code samples that meet the following requirements are

Reviewer 3: ```suggestion
  resources, only the code samples that meet the following requirements are
```

  typo or out-of-date content, you likely need to open multiple and subsequent
  PRs. If not a followup PR, at least add the "`cherrypick` labels" to your
  original PR to indicate in which of the past release that your change affects.

  For example, if you find a typo in a page of the `v0.5` release, then that
  page in the `master` branch likely also has that typo.

  To fix the typo:

  1.  Open a PR against the
      [`master`](https://github.com/knative/docs/tree/master/) branch.
  1.  Add one or more `cherrypick-#.#` labels to that PR to indicate which of
      the past release branches should also be fixed. Generally, we only
      maintain the most recent numbered release.
  1.  If you want to complete the fix yourself (**best practice**), you then open a
      subsequent PR by running `git cherry-pick [COMMIT#]` against the
      `release-0.5`.
      Where  `[COMMIT#]` is the commit of the PR that you merged in `master`.

      Note: Depending on workload and available bandwidth, one of the Knative
      team members might be able to help handle the `git cherry-pick` in order
      to push the fix into the affected release branch(es).

> See a list of the available documentaion versions in the

Reviewer: <!--typo-bot-->
Found potential misspelling(s):
```suggestion
See a list of the available documentation versions in the
```
